File: Cc.Ui/src/models/measure.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('measure')
export class Measure extends Entity {

    @field()
    id: number;
    @field()
    description: string;
    @field()
    name: string;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;
    @field()
    deleteUser: string;
    @field()
    deleteDate: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Measure {

        let item = new Measure();
        return item;
    }
}

@resource('measure')
export class MeasureSearchModel extends Entity implements Pagination {
    
    @field()
    description?: string;
    @field()
    name?: string;
    @field()
    deleteUser?: string;
    @field()
    deleteDate?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/Measure.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Measure
    {
        public Measure()
        {
            OrderRow = new HashSet<OrderRow>();
        }

        public long Id { get; set; }
        public string Description { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }
        public string? DeleteUser { get; set; }
        public DateTime? DeleteDate { get; set; }

        public virtual ICollection<OrderRow> OrderRow { get; set; }
    }
}
